src/PerformanceBenchmarkingUtilities.cxx

```diff
@@ -0,0 +1,169 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "PerformanceBenchmarkingInformation.h"
+#include "PerformanceBenchmarkingUtilities.h"
+#include <cstdlib>
+#include <ostream>
+#include <fstream>
+
+/**  Decorate with json from an environmental variable
+ *
+ * export ITKPERFORMANCEBENCHMARK_AUX_JSON="{ \"ITK_PROGRAMMERS_ARE\": [ \"Spectacular\", \"Awesome\", \"Brilliant\", \"Good Looking\" ] }"
+ *
+ * For ITK versions without the new itk::BuildInformation
+cd ${ITK_SRC}
+
+export GIT_CONFIG_SHA1="$(git rev-parse HEAD)"
+export GIT_CONFIG_DATE="$(git show -s --format=%ci HEAD)"
+export GIT_LOCAL_MODIFICATIONS="$(git  diff --shortstat HEAD)"
+export ITKPERFORMANCEBENCHMARK_AUX_JSON="
```

Comment:
@thewtex @dzenanz 

Check this out!  I'm pretty proud of it.  

We should be able to write a small script walk through all history of ITK, extract some meta-data, set this magic environmental variable, and run the performance tests going back in history to a point prior to having the itk::BuildInformation!

Reply:
When I get some free time, I'll be updating the project's main README.md to provide instructions on how to use these features, and add a script to assist with walking through histories.

Reply:
That would be awesome!

Reply:
Specular, brilliant, and awesome, @hjmjohnson ! And from a good looking fellow. I love it. :heart: :man_dancing: :man_cartwheeling: :zap:

Reply:
Wow, all the lately "revamping process", including these cool features, is leaving me speechless 🎆